spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -87,7 +94,8 @@ def argv(self):
                 CONF.set('main_interpreter', 'default', True)
                 CONF.set('main_interpreter', 'custom', False)
 
-        is_different_interpreter = pyexec != sys.executable
+        # Part of spyder-ide/spyder#11819
+        is_different_interpreter = validate_interpreter(pyexec)
```

Comment:
Is this correct? Should it ve Not ?

Reply:
This is like doing:
```
is_different_interpreter = (
    osp.dirname(pyexec) != osp.dirname(sys.executable) and 'python' in osp.basename(pyexec))
```
I think this gives the correct value but maybe I'm wrong @ccordoba12 @goanpeca ?

Reply:
It's fine as it is (that's how it was done before).